File: 0997-find-the-town-judge/0997-find-the-town-judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        int[] indeg = new int[n+1];
        for(int i = 0; i<trust.length; i++) {
            indeg[trust[i][0]]--;
            indeg[trust[i][1]]++;
        }

        for(int i = 1; i<=n; i++) {
            if(indeg[i] == n - 1) return i;
        }

        return -1;
    }
}

File: 0997-find-the-town-judge/0997-find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        vector<int> indeg(n+1, 0);
        for(auto it : trust) {
            indeg[it[0]]--;
            indeg[it[1]]++;
        }

        for(int i = 1; i<=n; i++) {
            if(indeg[i] == n - 1) return i;
        }

        return -1;
    }
};